autoload/buffer.vim

```diff
@@ -316,5 +321,11 @@ export def Completor(findstart: number, base: string): any
     if options.dup
         candidates->map((_, v) => v->extend({ dup: 1 }))
     endif
+
+    if candidates->empty()
```

Comment:
Shouldn't you also not check if 'base' this time has same prefix as previous 'base'?

Reply:
Can you split this into 2 PRs? They are separate issues.